src/lua-runtime-fastpm.lua

```diff
@@ -33,31 +33,57 @@ function schema.output_redshifts.action(output_redshifts)
 schema.declare{name='h',                 type='number', required=true, default=0.7, help="Dimensionless Hubble parameter"}
-schema.declare{name='pm_nc_factor',      type='array:number',  required=true,
-    help="A list of {a, PM resolution}. Add an entry {0.0, 1} to set lpt linear field resolution to 1."}
+schema.declare{name='pm_nc_factor',      type='array:number',  required=true, help="A list of {a, PM resolution}, "}
 schema.declare{name='np_alloc_factor',   type='number', required=true, help="Over allocation factor for load imbalance" }
 schema.declare{name='compute_potential', type='boolean', required=false, default=false, help="Calculate the gravitional potential."}
-schema.declare{name='m_ncdm',            type='array:number', required=false, default={}, help="Mass of ncdm particles. Enter in descending order."}
-schema.declare{name='every_ncdm',        type='number', required=false, default=4, help="Subsample ncdm from cdm every..."}
 schema.declare{name='n_shell',           type='number', required=false, default=10, help="Number of shells of FD distribution for ncdm splitting."}
-schema.declare{name='n_side',            type='number', required=false, default=2, help="Number of sides in HEALPix or fibonacci split. Note that this means n_fib for fibonacci."}
 schema.declare{name='lvk',               type='boolean', required=false, default=false, help="Use the low velocity kernel when splitting FD for ncdm."}
+schema.declare{name='n_side',            type='number', required=false, default=2, help="Number of sides in HEALPix split."}
+schema.declare{name='every_ncdm',        type='number', required=false, default=4, help="Subsample ncdm from cdm every..."}
 schema.declare{name='ncdm_sphere_scheme',type='enum', required=false, default="fibonacci", help="Split sphere with 'fibonacci' or 'healpix'?"}
 schema.ncdm_sphere_scheme.choices = {
     healpix = 'FASTPM_NCDM_SPHERE_HEALPIX',
     fibonacci = 'FASTPM_NCDM_SPHERE_FIBONACCI',
 }
+-- allow backward compatibility wth lowercase o
+function schema.omega_m.action (value)
+     if value ~= nil then
+         schema.Omega_m.default = value
+     end
+end
 
+-- check for bad input
 function schema.m_ncdm.action (m_ncdm)
+    if #m_ncdm ~= 0 then
+        function schema.T_cmb.action (T_cmb)
+            if T_cmb == 0 then
+                error("For a run with ncdm particles use T_cmb > 0 to include an ncdm background.")
+            end
+        end
+    end
+    
     for i=2, #m_ncdm do
         if m_ncdm[i] > m_ncdm[1] then
             error("Please input the heaviest ncdm particle first.")
         end
     end
 end
 
+-- Growth calculation --
+schema.declare{name='growth_mode', type='enum', default='LCDM', help='Evaluate growth factors using the popular LCDM-only approximations or with ODE with MD initial conditions.'}
+schema.growth_mode.choices = {
+    LCDM = 'FASTPM_GROWTH_MODE_LCDM',
+    ODE = 'FASTPM_GROWTH_MODE_ODE',
```

Comment:
Do we require ODE mode if neutrino or radiation is enabled in the background?
If so, perhaps also add a check in the parameter file. (like above).